### PR TITLE
Feature/password-change

### LINKED ISSUE #85: accounts: change password
We should have a way for the members to change their passwords. This would be a simple endpoint that checks authentication and takes the old and new passwords (the old one for extra verification). The rest would be similar to the register mechanism

### PR DESCRIPTION
**Pull request by coutinho21**
Closes #85 

The objective is to add a new endpoint to change an account's password.

# Review checklist
-   [ ] Properly documents API changes in `docs/openapi.yml`
-   [ ] Contains enough appropriate tests
-   [ ] Behavior is as expected
-   [ ] Clean, well structured code

### CODE REVIEW

File: src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt [hunk 1]
```diff
@@ -420,6 +422,45 @@ class AccountControllerTest @Autowired constructor(
...
+    inner class ChangePassword {
+        private val password = "test_password"
+        private val changePasswordAccount: Account = ObjectMapper().readValue(
+            ObjectMapper().writeValueAsString(testAccount),
+            Account::class.java
+        )
+
+        @BeforeEach
+        fun addAccount() {
+            changePasswordAccount.password = encoder.encode(changePasswordAccount.password)
```
Comment (bdmendes) [inline src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt]: Can this line be in a before all or init-like method? Only needs to be done once (and would actually fail if there were more test methods inside this inner class)

File: src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt [hunk 2]
```diff
@@ -420,6 +422,45 @@ class AccountControllerTest @Autowired constructor(
...
+            repository.save(changePasswordAccount)
+        }
+
+        @Test
+        fun `should change password`() {
```
Comment (bdmendes) [inline src/test/kotlin/pt/up/fe/ni/website/backend/controller/AccountControllerTest.kt:444]: You should also test the sad path, ie. the provided password is wrong and the request should fail.

**Review by bdmendes — CHANGES_REQUESTED** [review]

File: src/main/kotlin/pt/up/fe/ni/website/backend/dto/auth/ChangePassDto.kt [hunk 3]
```diff
@@ -0,0 +1,6 @@
+package pt.up.fe.ni.website.backend.dto.auth
+
+data class ChangePassDto(
```
Comment (ttoino) [inline src/main/kotlin/pt/up/fe/ni/website/backend/dto/auth/ChangePassDto.kt]: I think we should be consistent with our naming and always write out password in full

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -42,6 +42,7 @@ jobs:
         with:
           name: lint-reports
           path: build/reports/ktlint
+        if: always()
```
Comment (bdmendes) [inline .github/workflows/ci.yml]: Is this here by mistake?

**Review by bdmendes — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (coutinho21) [reply .github/workflows/ci.yml]: Yes I had to clone the project again and create the branch so when i pulled everything that i had from this pr, it added @ttoino's commits that i had locally and this branch didn't have remotely.

Comment (coutinho21) [reply .github/workflows/ci.yml]: Is it a problem?

Comment (bdmendes) [reply .github/workflows/ci.yml]: That PR was already merged, so this should not be here. Can you fix that?

**Review by bdmendes — APPROVED** [review]

**Review by ttoino — APPROVED** [review]

**Pull request merged**